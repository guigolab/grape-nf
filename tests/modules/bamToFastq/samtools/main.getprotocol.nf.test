nextflow_process {

    name "Test Process getProtocol"
    script "modules/bamToFastq/samtools/main.nf"
    process "getProtocol"

    test("Should get sequencing protocol from BAM file") {

        when {
            process {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments" ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
