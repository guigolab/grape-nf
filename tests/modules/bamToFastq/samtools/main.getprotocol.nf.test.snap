{
    "Should get sequencing protocol from BAM file": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "{\n\t\"general\": {\n\t\t\"protocol\": \"PairedEnd\",\n\t\t\"reads\": {\n\t\t\t\"total\": 1000,\n\t\t\t\"mapped\": {\n\t\t\t\t\"1\": 998,\n\t\t\t\t\"2\": 2\n\t\t\t},\n\t\t\t\"mappings\": {\n\t\t\t\t\"ratio\": 1.002,\n\t\t\t\t\"count\": 1002\n\t\t\t}\n\t\t},\n\t\t\"pairs\": {\n\t\t\t\"total\": 489,\n\t\t\t\"mapped\": {\n\t\t\t\t\"1\": 488,\n\t\t\t\t\"2\": 1\n\t\t\t},\n\t\t\t\"insert_sizes\": {\n\t\t\t\t\"74\": 1,\n\t\t\t\t\"102\": 1,\n\t\t\t\t\"104\": 1,\n\t\t\t\t\"107\": 1,\n\t\t\t\t\"111\": 2,\n\t\t\t\t\"112\": 1,\n\t\t\t\t\"113\": 2,\n\t\t\t\t\"123\": 1,\n\t\t\t\t\"124\": 2,\n\t\t\t\t\"125\": 1,\n\t\t\t\t\"126\": 3,\n\t\t\t\t\"127\": 2,\n\t\t\t\t\"129\": 2,\n\t\t\t\t\"133\": 3,\n\t\t\t\t\"135\": 2,\n\t\t\t\t\"137\": 2,\n\t\t\t\t\"141\": 2,\n\t\t\t\t\"143\": 2,\n\t\t\t\t\"144\": 1,\n\t\t\t\t\"150\": 1,\n\t\t\t\t\"152\": 2,\n\t\t\t\t\"153\": 3,\n\t\t\t\t\"154\": 2,\n\t\t\t\t\"155\": 1,\n\t\t\t\t\"156\": 2,\n\t\t\t\t\"157\": 1,\n\t\t\t\t\"158\": 1,\n\t\t\t\t\"159\": 1,\n\t\t\t\t\"160\": 4,\n\t\t\t\t\"161\": 5,\n\t\t\t\t\"162\": 3,\n\t\t\t\t\"163\": 4,\n\t\t\t\t\"164\": 8,\n\t\t\t\t\"165\": 4,\n\t\t\t\t\"166\": 7,\n\t\t\t\t\"167\": 6,\n\t\t\t\t\"168\": 11,\n\t\t\t\t\"169\": 9,\n\t\t\t\t\"170\": 12,\n\t\t\t\t\"171\": 4,\n\t\t\t\t\"172\": 3,\n\t\t\t\t\"173\": 6,\n\t\t\t\t\"174\": 13,\n\t\t\t\t\"175\": 12,\n\t\t\t\t\"176\": 9,\n\t\t\t\t\"177\": 12,\n\t\t\t\t\"178\": 12,\n\t\t\t\t\"179\": 11,\n\t\t\t\t\"180\": 15,\n\t\t\t\t\"181\": 10,\n\t\t\t\t\"182\": 11,\n\t\t\t\t\"183\": 11,\n\t\t\t\t\"184\": 13,\n\t\t\t\t\"185\": 10,\n\t\t\t\t\"186\": 14,\n\t\t\t\t\"187\": 10,\n\t\t\t\t\"188\": 10,\n\t\t\t\t\"189\": 8,\n\t\t\t\t\"190\": 4,\n\t\t\t\t\"191\": 4,\n\t\t\t\t\"192\": 11,\n\t\t\t\t\"193\": 11,\n\t\t\t\t\"194\": 9,\n\t\t\t\t\"195\": 11,\n\t\t\t\t\"196\": 10,\n\t\t\t\t\"197\": 8,\n\t\t\t\t\"198\": 3,\n\t\t\t\t\"199\": 5,\n\t\t\t\t\"200\": 8,\n\t\t\t\t\"201\": 5,\n\t\t\t\t\"202\": 7,\n\t\t\t\t\"203\": 5,\n\t\t\t\t\"204\": 6,\n\t\t\t\t\"205\": 5,\n\t\t\t\t\"206\": 2,\n\t\t\t\t\"208\": 2,\n\t\t\t\t\"209\": 6,\n\t\t\t\t\"210\": 4,\n\t\t\t\t\"211\": 2,\n\t\t\t\t\"212\": 1,\n\t\t\t\t\"213\": 2,\n\t\t\t\t\"214\": 3,\n\t\t\t\t\"215\": 2,\n\t\t\t\t\"216\": 1,\n\t\t\t\t\"217\": 2,\n\t\t\t\t\"218\": 2,\n\t\t\t\t\"219\": 4,\n\t\t\t\t\"220\": 1,\n\t\t\t\t\"221\": 2,\n\t\t\t\t\"222\": 1,\n\t\t\t\t\"223\": 1,\n\t\t\t\t\"224\": 2,\n\t\t\t\t\"233\": 1,\n\t\t\t\t\"235\": 2,\n\t\t\t\t\"237\": 1,\n\t\t\t\t\"7228\": 1,\n\t\t\t\t\"11975\": 1,\n\t\t\t\t\"20827\": 1,\n\t\t\t\t\"59774\": 1,\n\t\t\t\t\"59790\": 2,\n\t\t\t\t\"59793\": 1,\n\t\t\t\t\"59797\": 1,\n\t\t\t\t\"59799\": 1,\n\t\t\t\t\"59800\": 1,\n\t\t\t\t\"59805\": 1,\n\t\t\t\t\"59810\": 1,\n\t\t\t\t\"59812\": 1,\n\t\t\t\t\"59816\": 1,\n\t\t\t\t\"59817\": 1,\n\t\t\t\t\"59828\": 1,\n\t\t\t\t\"59829\": 1,\n\t\t\t\t\"59830\": 1,\n\t\t\t\t\"68769\": 1,\n\t\t\t\t\"80023\": 1\n\t\t\t}\n\t\t}\n\t}\n}"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T16:29:27.186894"
    }
}